pkg/networkservice/common/interpose/server.go

```diff
@@ -93,15 +97,15 @@ func (l *interposeServer) Request(ctx context.Context, request *networkservice.N
 		// Iterate over all cross connect NSEs to check one with passed state.
 
 		l.endpoints.Range(func(key string, crossNSEURL *url.URL) bool {
-			crossCTX := clienturlctx.WithClientURL(ctx, crossNSEURL)
+			crossCTX := extend.WithValuesFromContext(ctx, clienturlctx.WithClientURL(context.Background(), crossNSEURL))
 
 			// Store client connection and selected cross connection URL.
 			connInfo, _ = l.activeConnection.LoadOrStore(clientConnID, connectionInfo{
 				endpointURL:     clientURL,
 				interposeNSEURL: crossNSEURL,
 				requestingNSE:   true,
 			})
-			result, err = next.Server(crossCTX).Request(crossCTX, request)
+			result, err = l.connectServer.Request(crossCTX, request)
```

Comment:
I will prefer to set a context variable about direction, or extend client url to had information about direction was overridden.

Reply:
Let's continue to discuss here https://github.com/networkservicemesh/sdk/pull/576#discussion_r519550687